src/features/pagination/ui/Pagination.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useState, useEffect } from 'react';
+import { Button } from '@shared/ui/Button';
+import { Pagination as PaginationInterface } from '@shared/api/items';
+import { generateArray } from '@shared/lib/generateArray';
+import { Dropdown } from '@shared/ui/Dropdown';
+
+interface Props {
+  data: PaginationInterface;
+  currentPage: number;
+  setCurrentPage: (number: number) => void;
+  setItemsPerPage: (number: number) => void;
+  itemsPerPage: number;
+  onClickPrev?: () => void;
+  onClickNext?: () => void;
+}
+
+export const Pagination: FC<Props> = ({
+  data,
+  currentPage,
+  setCurrentPage,
+  setItemsPerPage,
+  itemsPerPage,
+  onClickPrev,
+  onClickNext,
+}) => {
+  const [pagesArray, setPagesArray] = useState<number[]>([]);
+  const [currentNumbers, setCurrentNumbers] = useState<number[]>([]);
+
+  const MAX_DISPLAYED_NUMBER = 5;
+  const { last_visible_page: lastPage } = data;
+
+  useEffect(() => {
+    const array = generateArray(lastPage);
+    setPagesArray(array);
+  }, [lastPage]);
+  useEffect(() => {
+    setCurrentNumbers(pagesArray.slice(0, MAX_DISPLAYED_NUMBER));
+  }, [pagesArray]);
+
+  const renderMiddlePartOfPagination = () => {
+    return currentNumbers.map((pageNumber) => (
+      <Button
+        key={pageNumber}
+        onClick={() => setCurrentPage(pageNumber)}
+        style={{ fontWeight: pageNumber === currentPage ? 'bold' : 'normal' }}
```

Comment:
can we avoid using styles in markup?
read about ** clsx** package. [click](https://www.npmjs.com/package/clsx)
using this library you can easily rewrite this code with something like:
```suggestion
        classname={clsx(styles.pageButtonBase, { styles.pageButtonSelected: pageNumber === currentPage })}
```
That will help you to separate **markup** and **styles**